src/dotnet/SlnProjectExtensions.cs

```diff
@@ -12,17 +12,29 @@ public static class SlnProjectExtensions
     {
         public static IList<string> GetSolutionFoldersFromProject(this SlnProject project)
         {
-            var currentDirString = $".{Path.DirectorySeparatorChar}";
+            var solutionFolders = new List<string>();
 
-            var directoryPath = Path.GetDirectoryName(project.FilePath);
-            if (directoryPath.StartsWith(currentDirString))
+            var projectFilePath = project.FilePath;
+            if (!projectFilePath.StartsWith(".."))
             {
-                directoryPath = directoryPath.Substring(currentDirString.Length);
+                var currentDirString = $".{Path.DirectorySeparatorChar}";
+                if (projectFilePath.StartsWith(currentDirString))
+                {
+                    projectFilePath = projectFilePath.Substring(currentDirString.Length);
+                }
+
+                var projectDirectoryPath = Path.GetDirectoryName(projectFilePath);
+                if (!string.IsNullOrEmpty(projectDirectoryPath))
```

Comment:
Why do we call Path.GetDirectoryName twice? Once here and once again inside the if?

I think in general we would benefit from some aptly named methods that would document the intent of this code.